src/button-group/interfaces.ts

```diff
@@ -0,0 +1,121 @@
+   * * `text` (string) - The name shown as a tooltip or menu text for this button.
+   * * `disabled` (optional, boolean) - The disabled state indication for the button.
+   * * `loading` (optional, boolean) - The loading state indication for the button.
+   * * `loadingText` (optional, string) - The loading text announced to screen-readers.
+   * * `iconName` (optional, string) - Specifies the name of the icon, used with the [icon component](/components/icon/).
+   * * `iconAlt` (optional, string) - Specifies alternate text for the icon when using `iconUrl`.
+   * * `iconUrl` (optional, string) - Specifies the URL of a custom icon.
+   * * `iconSvg` (optional, ReactNode) - Custom SVG icon. Equivalent to the `svg` slot of the [icon component](/components/icon/).
+   * * `popoverFeedback` (optional, string) - Text that appears when the user clicks the button. Use to provide feedback to the user.
+   *
+   * ### menu-dropdown
+   *
+   * * `id` (string) - The unique identifier of the button, used to focus the button using `ref.focus(id)`.
+   * * `text` (string) - The name of the menu button shown as a tooltip.
+   * * `disabled` (optional, boolean) - The disabled state indication for the menu button.
+   * * `loading` (optional, boolean) - The loading state indication for the menu button.
+   * * `loadingText` (optional, string) - The loading text announced to screen-readers.
+   * * `items` (ButtonDropdownProps.ItemOrGroup[]) - The array of dropdown items that belong to this menu.
+   *
+   * group
+   *
+   * * `text` (string) - The name of the group rendered as ARIA label for this group.
+   * * `items` ((ButtonGroupProps.IconButton | ButtonGroupProps.MenuDropdown)[]) - The array of items that belong to this group.
+   */
+  items: ReadonlyArray<ButtonGroupProps.ItemOrGroup>;
+  /**
+   * Called when the user clicks on an item, and the item is not disabled. The event detail object contains the id of the clicked item.
+   */
+  onItemClick?: NonCancelableEventHandler<ButtonGroupProps.ItemClickDetails>;
+}
+
+export interface InternalButtonGroupProps extends ButtonGroupProps, InternalBaseComponentProps {}
+
+export namespace ButtonGroupProps {
+  // Fixes documenter output.
+  // When a union only has one element the resulting type for it is not string but object.
+  export type Variant = 'icon' | 'icon';
+
+  export type ItemOrGroup = Item | Group;
+  export type Item = IconButton | MenuDropdown;
+
+  export interface IconButton {
+    type: 'icon-button';
+    id: string;
+    text: string;
+    disabled?: boolean;
+    loading?: boolean;
+    loadingText?: string;
+    iconName?: IconProps.Name;
+    iconAlt?: string;
+    iconUrl?: string;
+    iconSvg?: React.ReactNode;
+    popoverFeedback?: React.ReactNode;
+  }
+
+  export interface MenuDropdown {
+    type: 'menu-dropdown';
+    id: string;
+    text: string;
+    disabled?: boolean;
```

Comment:
We recently introduced the disabled with reason feature (https://github.com/cloudscape-design/components/pull/2409). I can imagine we'll get some why they are not available on the ButtonGroup component.

Is there a specific reason we do not add the disabled reason here as well?

Reply:
Yes, the disabled reasons are not supported for icon buttons as of yet.